src/Centreon/Domain/PlatformTopology/PlatformTopology.php

```diff
@@ -0,0 +1,268 @@
+    private const AVAILABLE_TYPES = [
+        self::TYPE_CENTRAL,
+        self::TYPE_POLLER,
+        self::TYPE_REMOTE,
+        self::TYPE_MAP,
+        self::TYPE_MBI
+    ];
+
+    /**
+     * @var int Id of server
+     */
+    private $id;
+
+    /**
+     * @var string|null name
+     */
+    private $name;
+
+    /**
+     * @var string|null Server type
+     */
+    private $type;
+
+    /**
+     * @var string|null Server address
+     */
+    private $address;
+
+    /**
+     * @var string|null Server parent address
+     */
+    private $parentAddress;
+
+    /**
+     * @var int|null Server parent id
+     */
+    private $parentId;
+
+    /**
+     * @var int|null Server nagios ID for Central only
+     */
+    private $serverId;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int|null $id
+     * @return $this
+     */
+    public function setId(?int $id): self
+    {
+        $this->id = $id;
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getType(): ?string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @param string|null $type server type: central, poller, remote, map or mbi
+     *
+     * @return $this
+     * @throws \InvalidArgumentException
+     */
+    public function setType(?string $type): self
+    {
+        $type = strtolower($type);
+
+        // Check if the server_type is available
+        if (!in_array($type, static::AVAILABLE_TYPES)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    _("The type of platform '%s'@'%s' is not consistent"),
+                    $this->getName(),
+                    $this->getAddress()
+                )
+            );
+        }
+        $this->type = $type;
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string|null $name
+     * @return $this
+     * @throws \InvalidArgumentException
+     */
+    public function setName(?string $name): self
+    {
+        $name = filter_var($name, FILTER_SANITIZE_STRING);
+        if (empty($name)) {
+            throw new \InvalidArgumentException(
+                _('The name of the platform is not consistent')
+            );
+        }
+        $this->name = $name;
+        return $this;
+    }
+
+    /**
+     * Validate address consistency
+     *
+     * @param string|null $address the address to be tested
+     *
+     * @return string|null
+     * @throws \InvalidArgumentException
+     */
+    private function checkIpAddress(?string $address): ?string
+    {
+        if ($address === null) {
+            return $address;
+        }
+
+        // Check for valid IPv4 or IPv6 IP
+        if (false !== filter_var($address, FILTER_VALIDATE_IP)) {
+            return $address;
+        }
```

Comment:
```suggestion
        if ($address === null || false !== filter_var($address, FILTER_VALIDATE_IP)) {
            return $address;
        }
```

Reply:
did an $address could be null ?  

As you remove the "null" in your json validator maybe you could also remove the ?string for string typehint , and don't check if $address === null ?

Reply:
I restored the null option in json_validator